hexapod_description/urdf/modules/hexaforce.xacro

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+
+  <!-- XACRO PROPERTIES -->
+  <xacro:property name="PI" value="3.14159265359" />
+  <xacro:property name="PI_2" value="1.57079632679" />
+  <xacro:property name="PI_4" value="0.78539816339" />
+  <xacro:property name="scale" value="0.001" />
+  <xacro:property name="body_mass" value="1.031"/>
+  <xacro:property name="body_length" value="0.24"/>
+  <xacro:property name="body_width" value="0.119"/>
+  <xacro:property name="body_width_extra" value="0.199"/>
+  <xacro:property name="body_height" value="0.04"/>
+
+  <xacro:property name="leg1_width" value="0.044"/>
+  <xacro:property name="leg1_length" value="0.06"/>
+  <xacro:property name="leg1_dist" value="0.2"/>
+  <xacro:property name="leg1_mass" value="0.02"/>
+
+  <xacro:property name="leg2_width" value="0.038"/>
+  <xacro:property name="leg2_length" value="0.0835"/>
+  <xacro:property name="leg2_dist" value="0.2"/>
+  <xacro:property name="leg2_mass" value="0.184"/>
+
+  <xacro:property name="leg3_width" value="0.054"/>
+  <xacro:property name="leg3_length" value="0.065"/>
+  <xacro:property name="leg3_height" value="0.10"/>
+  <xacro:property name="leg3_tip_radius" value="0.02"/>
+  <xacro:property name="leg3_mass" value="0.04"/>
+
+  <xacro:property name="force_y_offset" value="${(11.25+27)*0.001}"/>
+  <xacro:property name="force_z_offset" value="0.115"/>
+
+  <xacro:property name="joint_lim" value="${PI_2}"/>
+  <xacro:property name="joint_eff" value="30.0"/>
```

Comment:
I think this joint maximum force is huge! We should set it to something that is closer to the actual motors. Which dynamixels are we using the hexapods?

Reply:
We are using MX-28AT. The documentation says we can expect 2.5N.m at 12V (recommended voltage). Is N.m the unit in URDF.

Reply:
> Is N.m the unit in URDF?

Yes (for rotational joints), but the `effort` limit is the limit of the total force that can be applied to the joint (including things coming from gear ratios etc). Is the 2.5Nm this value? I guess it is (it says stall torque in the specs). Of course, we need to set the appropriate `damping` and `friction` for the joint dynamics.. I will try to do this..

Reply:
I believe the force given by the manufacturer is indeed the actuator's output, after all internal reductions. I don't find any damping nor friction information on Robotis' website. However, some persons have already done characterisations for these actuators.